harvester/src/test/java/com/github/reflectoring/infiniboard/harvester/source/config/UpdatePluginConfigJobTest.java

```diff
@@ -0,0 +1,95 @@
+package com.github.reflectoring.infiniboard.harvester.source.config;
+
+import com.github.reflectoring.infiniboard.harvester.scheduling.SchedulingService;
+import com.github.reflectoring.infiniboard.packrat.source.SourceConfig;
+import com.github.reflectoring.infiniboard.packrat.widget.WidgetConfig;
+import com.github.reflectoring.infiniboard.packrat.widget.WidgetConfigRepository;
+import org.junit.Test;
+import org.quartz.JobKey;
+import org.quartz.SchedulerException;
+import org.springframework.context.ApplicationContext;
+
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+
+import static org.mockito.Mockito.*;
+
+/**
+ * Created by steven on 11.06.16.
+ */
+public class UpdatePluginConfigJobTest {
+
+    @Test
+    public void remembersModified() {
+        ApplicationContext applicationContext = mock(ApplicationContext.class);
+        SchedulingService schedulingService = mock(SchedulingService.class);
+        WidgetConfigRepository repository = mock(WidgetConfigRepository.class);
+        when(applicationContext.getBean(SchedulingService.class)).thenReturn(schedulingService);
+        when(applicationContext.getBean(WidgetConfigRepository.class)).thenReturn(repository);
+
+        UpdatePluginConfigJob job = new UpdatePluginConfigJob();
+        job.executeInternal(applicationContext, new JobKey("update", "harvester"), Collections.emptyMap());
+        job.executeInternal(applicationContext, new JobKey("update", "harvester"), Collections.emptyMap());
+        job.executeInternal(applicationContext, new JobKey("update", "harvester"), Collections.emptyMap());
+
+        // first call searches all configurations
+        verify(repository, atMost(1)).findAll();
```

Comment:
we should discuss this technique of assertions. looks like you assert that an internal behavior is done in a specific way. but you should black-box test the method, so the internal implementation can be refactored, without changing the tests.